sk-usbhid.c

```diff
@@ -880,7 +880,7 @@ sk_enroll(uint32_t alg, const uint8_t *challenge, size_t challenge_len,
 	}
 	if (device != NULL)
 		sk = sk_open(device);
-	else
+	else if ((sk = sk_open("windows://hello")) == NULL)
```

Comment:
Thinking about this a bit more, maybe the `sk_open("windows://hello")` call should move inside `sk_probe()`? It looks like we only want to open WinHello when we're about to fall back to probing anyway

Reply:
We don't want to `sk_open("windows://hello")` from sk_load_resident_keys(), as the underlying functionality is not yet supported by Microsoft's webauthn.dll. Adding a `prefer_winhello` parameter to sk_probe() would give the function enough context to decide whether to prefer `windows://hello` should `sk_open("windows://hello")` succeed.

Reply:
What would WinHello's behaviour be in this case? Would it just fail with it as an unsupported operation?

If so, that's IMO okay because the behaviour on Cygwin would be consistent for all uses of FIDO keys: OpenSSH would use WinHello unless explicitly passed a -Odevice to use.

Of course, if WinHello behaves badly when asked to return resident keys then this would be a non-starter :)

Reply:
To be explicit: if this is workable, then `sk_probe()` wouldn't need the `prefer_winhello` option

Reply:
WinHello would cause fido_credman_get_dev_metadata(), fido_credman_get_dev_rp(), and fido_credman_get_dev_rp() (used by read_rks() in sk-usbhid.c) to reliably fail with FIDO_ERR_INTERNAL. While workable, defaulting to WinHello in this case seems incongruous with the way `ssh-keygen -K` works on every other platform.

Users will already need to issue `ssh-keygen -K` from an administrator prompt, but that is a circumstance beyond our control. Asking them to provide a device path (which effectively makes `ssh-keygen -K` dependent on `fido2-token -L` or a similar tool) when we could handle the operation in harmony with the way it is implemented in other platforms, seems like an unnecessary hindrance to me.

To be clear: I have no strong feelings about this. :) Just offering a different perspective. On a side note, it might be worth looking at https://github.com/openssh/openssh-portable/pull/329 for further PIN-prompting improvements (I will ask @LDVG to mark the PR as ready when they return from vacation).